src/tools/create_ontology_definition_lookup_table.py

```diff
@@ -0,0 +1,83 @@
+"""
+Gathers metadata of ontologies found in the OLS api and adds any ontologies 
+necessary that are not included in ols.
+Jira ticket FD-1381
+"""
+
+import requests
+import pandas as pd
+import sys
+import os
+
+# Specify the path to resources
+sys.path.append(os.path.abspath(os.path.join(os.path.dirname(__file__), '..')))
+
+from locutus_util.resources import ONTOLOGY_API_LOOKUP_TABLE_PATH, OLS_API_BASE_URL, MONARCH_API_BASE_URL, LOINC_API_BASE_URL
+
+JIRA_ISSUES = ['fd1381']
+
+# Define the api endpoints used to request ontologies
+ols_ontologies_url = f"{OLS_API_BASE_URL}ontologies"
+
+# Initialize an empty list to store the extracted data
+extracted_data = []
+
+# Function to fetch data from a given URL
+def fetch_data(url):
+    response = requests.get(url)
+    if response.status_code == 200:
+        return response.json()
+    else:
+        print(f"Failed to fetch data: {response.status_code}")
+        return None
+    
+
+# Fetch the first page of data
+print("Fetching data")
+data = fetch_data(ols_ontologies_url)
+# Loop through the pages until there are no more pages left
+print("Transforming data")
+while data:
+    ontologies = data['_embedded']['ontologies']
+    for ontology in ontologies:
+        extracted_data.append({
+            'api_url': OLS_API_BASE_URL,
+            'api_source': 'ols',
+            'ontology_code': ontology['ontologyId'],
+            'curie': ontology['config'].get('preferredPrefix', 'N/A'),
```

Comment:
So, for things like this, how I handle missingness will differ depending what is consuming the output. For instance, if completeness is vital, then I would continue with your current approach, though, I would probably replace 'N/A' with either None or '', depending on which is better suited for the final destination (if R or Pandas is the final destination, N/A may actually be the best choice). For a lot of things, however, it isn't necessary for inappropriate fields to be present with an indication that it doesn't apply. I've typically handled those by adding them into the "extracted_data" object only if they are there. 

Also, I think it's OK if we establish some of those pieces as being "required" and raising an exception if that doesn't exist. "Title" and "System" seem like they should be required to me. 

All of that said, Curie may be a special case where it is required but maybe it can also be ''. I think we'll need that information to properly identify codes as they come in from the searches. But, in the case of LOINC, there won't be one, since we will probably be pulling from LOINC itself.

Reply:
Unfortunately, 'Title', 'System', and 'Curie' are not always present when using the OLS API. I'll give the data another look. Maybe there are other fields that are sometimes used instead.

Reply:
For those that are incomplete, I wonder how important they might be. Could you pull together a quick summary of how many have issues like that and put together a list of what ontologies that are impacted. I could imagine we could supplement them if the number isn't too high, or we could supplement the important ones and then let the others that are incomplete just go away with a document that indicates which are affected in case we want to add supplementary details in the future.

Reply:
The request was completed via Slack. PR on hold until decisions are made.